internal/venice-common/src/main/java/com/linkedin/venice/init/AbstractSystemSchemaInitializer.java

```diff
@@ -0,0 +1,299 @@
+package com.linkedin.venice.init;
+
+import static com.linkedin.venice.ConfigKeys.CONTROLLER_SYSTEM_SCHEMA_CLUSTER_NAME;
+
+import com.linkedin.venice.common.VeniceSystemStoreType;
+import com.linkedin.venice.controllerapi.UpdateStoreQueryParams;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.schema.SchemaData;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.writecompute.WriteComputeSchemaConverter;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.utils.Utils;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.avro.Schema;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+
+/**
+ * This class provides the basic framework necessary to create new schema system stores and register previously unknown
+ * schemas for schema system stores.
+ */
+public abstract class AbstractSystemSchemaInitializer {
```

Comment:
> @nisargthakkar Can you put a list of deployment ordering issues this PR will address? I would like to see which ones are addressed to make sure we cover the full scope. From my understanding, there are several deployment ordering issues:
> 
> 1. Child Controller needs to be deployed first before Parent Controller whenever there is a KME/AdminOperation protocol evolution.
> 2. The backend needs to be deployed first before VPJ/Samza Job whenever there is KME protocol upgrade.
> 3. Venice Repush needs to be able to deserialize the events produced by Venice Server, and during KME protocol upgrade, Repush may have an issue since it is not using a SchemaReader to find out the unknown schemas.
> 4. For AA stores, if there is a KME upgrade, we need to make sure Controller in each colo registers the new KME protocol before any component starts using it.
> 
> Can you share whether this PR will cover all of them or only part of them?

[Hopefully this let's us continue the conversation in a thread]

Hi @gaojieliu. Good points. I updated the PR description about the deployment orderings

I had completely forgotten about the active/active KME use cases. This is kind of tricky to implement since we have a SchemaPresenceChecker only for the local colo. We will probably need to add a parent controller end point to help check if all child colos have registered a protocol version

Will address the current PR review comments and think about this more. In the meantime, please share your thoughts in this thread (if github creates one)

Reply:
The implementation in this PR already invokes parent controller endpoints to register schemas site-wide. What it currently does not do is wait for the schema to percolate to all the child colos. I plan to add a parent controller endpoint that will query the child controllers of all colos to confirm if the schema got registered. 

I realized yesterday that there is another miss, though. In the current code, Leader child controller will register a new schema locally - but follower child controllers will register them with parent controller. Will try to see how I can unify this

Reply:
Hey @gaojieliu. I was thinking about integration tests and by registering the schemas with parent controller leads to a cyclic dependency issue.

So, I have updated the initialization to be local-colo specific. However, I feel the framework we have now can handle the registration of schemas with parent controller once we figure out how to handle the cyclic dependency issue.